File: software/lamp-ui/src/utils/stringCase.ts
// Title-cases each word: first letter upper, rest lower. Lamps store names
// canonically lowercase; this is the display form. `jacko` -> `Jacko`,
// `living room` -> `Living Room`.
export function toTitleCase(s: string): string {
  return s.toLowerCase().replace(/\b\w/g, (c) => c.toUpperCase())
}

File: software/lamp-app-flutter/lib/core/utils/string_case.dart
extension TitleCase on String {
  /// Title-cases each word: first letter upper, rest lower, whitespace kept.
  /// Lamps store names canonically lowercase; this is the display form.
  /// `jacko` -> `Jacko`, `living room` -> `Living Room`, `` -> ``.
  String toTitleCase() => toLowerCase()
      .replaceAllMapped(RegExp(r'\b\w'), (m) => m[0]!.toUpperCase());
}
